ext/OceananigansAMDGPUExt/Grids.jl

```diff
@@ -0,0 +1,10 @@
+module Grids
+
+using AMDGPU
+using ..Architectures: ROCmGPU
+
+import Base: zeros
+
+zeros(FT, ::ROCmGPU, N...) = AMDGPU.zeros(FT, N...)
+
+end # module
```

Comment:
We can remove all this extension by using KernelAbstractions instead

Reply:
@navidcy - what do you think here ? You had put in some effort to moving the AMDGPU bits into the extension. Would you also be in favor of using KernelAbstractions instead ? @vchuravy do you foresee any potential limitations for future development, should we take the approach of using KernelAbstractions ?

I suspect that @simone-silvestri is hinting at using KernelAbstractions in all other places; correct me if I'm wrong here

Reply:
I would welcome contributions to KA to add any missing functionality.